### PR TITLE
Removing the use of the ioutil package And Fix CVE-2021-42576

### PR DESCRIPTION
**Pull request by qiangmzsx**
# Description

1. Fix formation security vulnerability in [CVE-2021-42576](https://nvd.nist.gov/vuln/detail/CVE-2021-42576).
2. The current minimum version of dapr is golang 1.18, which is no longer compatible with the previous version and can remove the marked deprecated functions.
## Checklist

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by berndverst** [comment]
I'll take a look at this PR today. What is the impact of this CVE?

File: middleware/http/nethttpadaptor/nethttpadaptor.go [hunk 1]
```diff
@@ -29,11 +29,11 @@ func NewNetHTTPHandlerFunc(logger logger.Logger, h fasthttp.RequestHandler) http
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		c := fasthttp.RequestCtx{}
 		remoteIP := net.ParseIP(r.RemoteAddr)
-		remoteAddr := net.IPAddr{remoteIP, ""} //nolint
+		remoteAddr := net.IPAddr{remoteIP, ""} // nolint
```
Comment (ItalyPaleAle) [inline middleware/http/nethttpadaptor/nethttpadaptor.go:32]: Please keep the space here. Comments meant for machines need not to have spaces after `//`

**Review by ItalyPaleAle — DISMISSED** [review]
Aside from one minor thing, LGTM! Thanks :)

File: middleware/http/nethttpadaptor/nethttpadaptor.go (hunk 1 above)
Comment (berndverst) [reply middleware/http/nethttpadaptor/nethttpadaptor.go:32]: Just use the suggestion mode and then you can directly commit the suggestion from the UI here @ItalyPaleAle

Comment (berndverst) [reply middleware/http/nethttpadaptor/nethttpadaptor.go:32]:
```suggestion
		remoteAddr := net.IPAddr{remoteIP, ""} //nolint
```

**Review by berndverst — APPROVED** [review]

**Review by pkedy — APPROVED** [review]

File: middleware/http/nethttpadaptor/nethttpadaptor.go (hunk 1 above)
Comment (ItalyPaleAle) [reply middleware/http/nethttpadaptor/nethttpadaptor.go]: I don't have that option :( Probably because I'm not a maintainer

**Review by ItalyPaleAle — APPROVED** [review]

**Review by mukundansundar — APPROVED** [review]

**Review by yaron2 — APPROVED** [review]

**Comment by berndverst** [comment]
FWIW we still do not know what impact this actually has on Dapr.

**Comment by berndverst** [comment]
```
go mod why github.com/microcosm-cc/bluemonday
# github.com/microcosm-cc/bluemonday
github.com/dapr/components-contrib/middleware/http/oauth2
github.com/fasthttp-contrib/sessions
github.com/fasthttp-contrib/sessions.test
github.com/kataras/go-serializer
gopkg.in/kataras/go-serializer.v0/markdown
github.com/microcosm-cc/bluemonday
```

**Comment by berndverst** [comment]
Likely the CVE impact is limited to the OAuth2 middleware (if present at all)

**Pull request merged**